src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -116,15 +116,17 @@ let run ~logger ~trust_system ~verifier ~network ~time_controller
   Ledger_catchup.run ~logger ~precomputed_values ~trust_system ~verifier
     ~network ~frontier ~catchup_job_reader ~catchup_breadcrumbs_writer
     ~unprocessed_transition_cache ;
-  Strict_pipe.Reader.iter_without_pushback clear_reader ~f:(fun _ ->
-      let open Strict_pipe.Writer in
-      kill valid_transition_writer ;
-      kill primary_transition_writer ;
-      kill processed_transition_writer ;
-      kill catchup_job_writer ;
-      kill catchup_breadcrumbs_writer ;
-      if Ivar.is_full clean_up_catchup_scheduler then
-        [%log error] "Ivar.fill bug is here!" ;
-      Ivar.fill clean_up_catchup_scheduler () )
-  |> don't_wait_for ;
+  upon (Strict_pipe.Reader.read clear_reader) (function
+    | `Eof ->
+        [%log error] "unexpected eof on clear_reader"
+    | `Ok _ ->
+        let open Strict_pipe.Writer in
+        kill valid_transition_writer ;
```

Comment:
Shouldn't we `kill` the pipes regardless? It should be safe either way.

Reply:
I don't know how the first case could ever trigger, honestly, so I'm fine to proceed with your suggestion @mrmr1993